tests/rptest/tests/nodes_decommissioning_test.py

```diff
@@ -547,20 +547,24 @@ def test_flipping_decommission_recommission(self):
             # from finishing decommission operation
             self._set_recovery_rate(64)
             admin.decommission_broker(id=node_id)
-            wait_until(lambda: self._partitions_moving(node=survivor_node),
+            wait_until(lambda: self._partitions_moving(node=survivor_node) or
+                       self._node_removed(node_id, survivor_node),
                        timeout_sec=15,
                        backoff_sec=1)
+            if self._node_removed(node_id, survivor_node):
```

Comment:
I don't quite understand - if we prevent the node decom by throttling recovery, how did it finish?